### PR TITLE
rz-ghidra, jsdec: init at 0.6.0; rizin-sigdb: init at unstable-2023-02-13

### PR DESCRIPTION
**Pull request by chayleaf**
###### Description of changes

Related to #86448 (which is about Radare2, which Rizin forks from)

Inits https://github.com/rizinorg/rz-ghidra/ and https://github.com/rizinorg/jsdec/

This adds the ability to do the following:

```nix
rizin.withPlugins (ps: with ps; [ jsdec rz-ghidra sigdb ])
cutter.withPlugins (ps: with ps; [ jsdec rz-ghidra sigdb ])
```

It does this by creating a "portable" Rizin build (which looks for files relative to `/proc/self/exe`), but overrides the logic to use `NIX_RZ_PREFIX` env var instead.

Also, since Cutter looks for plugins in `XDG_DATA_DIRS` , the wrapper has to extend `XDG_DATA_DIRS` as well.

Sigdb isn't a real plugin, it's a signature database for Rizin that allows restoring some symbols. It's supposed to be downloaded during Rizin build, but that makes no sense to do in general as it's quite huge (50MB) and making it toggleable using a derivation flag would cause a Rizin rebuild.

TODO: rz-retdec (currently retdec package is outdated by 2 major versions and is missing e.g. cmake config, so rz-retdec can't be built reproducibly)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@ofborg eval

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2587,6 +2587,13 @@
     githubId = 46527539;
     name = "ChaosAttractor";
   };
+  chayleaf = {
```
Comment (Mic92) [inline maintainers/maintainer-list.nix:2590]: When updating this file to solve the merge conflict notice that we also now have a ci check to ensure the file is kept alphabetical sorted.

**Comment by chayleaf** [comment]
seeing as it seemingly won't compile on darwin, should I add `broken = stdenv.isDarwin`? I don't have a Mac so I can't fix this

**Comment by 7c6f434c** [comment]
> `broken = stdenv.isDarwin`

Actually, do I understand correctly that `platforms` is not set at all?

**Comment by XVilka** [comment]
Now there are changes for plugins - a new additional PREFIX for loading plugins. Might be useful for you: https://github.com/rizinorg/rizin/blob/dev/.github/workflows/ci.yml#L766

cc @ret2libc

**Comment by chayleaf** [comment]
it isn't, extra_prefix is hardcoded at build time, it would be bad to rebuild rizin for each different set of plugins. And of course I could replace it with a fake path much like CA derivations do and then instead of wrapping rizin I could copy the binaries and replace the path within them, but that isn't actually needed anyway because rizin always looks plugins up at a path relative to the binaries.

**Pull request merged**